learning_observer/learning_observer/rosters.py

```diff
@@ -413,6 +419,9 @@ async def courselist(request):
     if settings.pmss_settings.source(types=['roster_data']) in ["google_api"]:
         runtime = learning_observer.runtime.Runtime(request)
         return await learning_observer.google.courses(runtime)
+    elif settings.pmss_settings.source(types=['roster_data']) in ["canvas"]:
+        runtime = learning_observer.runtime.Runtime(request)
+        return await learning_observer.canvas.courses(runtime)
```

Comment:
This code and the changes below are **super** similar to the Google code. How can it be abstracted? We want more LMSs, but we don't want to repeat the same 3 lines over and over again.

Reply:
Fixed by adding handler function maps